moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -348,12 +352,12 @@ void PlanningSceneMonitor::scenePublishingThread()
     ROS_DEBUG_NAMED(LOGNAME, "Published the full planning scene: '%s'", msg.name.c_str());
   }
 
+  ros::Rate rate(publish_planning_scene_frequency_);
   do
   {
     moveit_msgs::PlanningScene msg;
     bool publish_msg = false;
     bool is_full = false;
-    ros::Rate rate(publish_planning_scene_frequency_);
```

Comment:
isn't this recreated every iteration and thus reacts to rate changes?

Reply:
Good point, I think my initial bug was I was not adjust the PSM rate that my widget was visualizing. I think there are still some useful improvements in this PR though (using atomic). I will update the description though.